### PR TITLE
github: add workflow that allows us to pin dependencies

### PR DESCRIPTION
**Pull request by carlaKC**
This PR adds a workflow that allows us to pin certain dependencies that we do not want to upgrade. 

Not _certain_ that we want to do this, but I thought it could be useful to prevent accidental upgrade?

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -280,3 +280,24 @@ jobs:
...
+        pinned_dep:
+          - google.golang.org/grpc v1.24.0
+          - github.com/golang/protobuf v1.3.2
+
+    steps:
+      - name: git checkout
+        uses: actions/checkout@v2
+
+      - name: ensure dependences at correct version
+        run: if [ -z "$(cat go.mod | grep "${{ matrix.pinned_dep }}")" ]; then echo dependency ${{ matrix.pinned_dep }} should not be altered ; exit 1 ; fi
```
Comment (guggero) [inline .github/workflows/main.yml:303]: I think the if can be simplified to: `if ! grep -q "${{ matrix.pinned_dep }}" go.mod; then ...`.
The `-q` instructs grep to just return a 0 exit code if there was a match.

**Review by guggero — APPROVED** [review]
I think it's a good idea! That way contributors have a chance at fixing this on their own if they see the check fail.
It's also a very cheap test to run, so I think we should add this. LGTM :heavy_check_mark:

**Review by bhandras — APPROVED** [review]
Nice! :+1:

**Comment by Roasbeef** [comment]
Would these be deps used to run the itests (like bitcoind) version, or deps at the go.mod/sum level?

**Comment by carlaKC** [comment]
> deps at the go.mod/sum level?

go.mod/sum level, aiming to catch the case where there's an accidental upgrade (happens occasionally in contributor PRs). 

> Would these be deps used to run the itests (like bitcoind) version

Not covering these as is, because it's easier to catch the explicit upgrade in review. Whereas mod/sum changes are a bit easier to miss IMO.

**Pull request merged**